components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/AuthorizationHandlerManager.java

```diff
@@ -120,10 +125,43 @@ private OAuth2AuthorizeRespDTO validateAuthzRequest(OAuth2AuthorizeReqDTO authzR
         if (isInvalidAccessDelegation(authzReqDTO, authorizeRespDTO, authzReqMsgCtx, authzHandler)) {
             return authorizeRespDTO;
         }
-        validateScope(authzReqDTO, authorizeRespDTO, authzReqMsgCtx, authzHandler);
+
+        //Execute Internal SCOPE Validation.
+        JDBCPermissionBasedInternalScopeValidator scopeValidator = new JDBCPermissionBasedInternalScopeValidator();
+        String[] authorizedInternalScopes = scopeValidator.validateScope(authzReqMsgCtx);
+        //Clear the internal scopes. Those scopes not required for scope validations.
```

Comment:
We can make this comment more understandable. Since before this line, there is a scope validation logic it is hard to understand